packages/pwa-buildpack/lib/WebpackTools/configureWebpack/getClientConfig.js

```diff
@@ -199,22 +199,24 @@ async function getClientConfig(opts) {
         }
     } else if (mode === 'production') {
         let versionBanner = '';
-        try {
-            versionBanner = projectConfig.section('staging').buildId;
-            if (!versionBanner || versionBanner.trim().length === 0) {
-                throw new Error('invalid build id');
-            }
-        } catch (error) {
-            try {
-                versionBanner = require('child_process')
-                    .execSync('git describe --long --always --dirty=-dev')
-                    .toString();
-            } catch (e) {
-                versionBanner = `${
-                    require(path.resolve(context, './package.json')).version
-                }-[hash]`;
-            }
-        }
+        const packageJson = require(path.resolve(context, './package.json'));
+        const packageRegex = /^@magento|^@apollo/;
+        const pwaStudioVersions = {
+            'pwa-studio': packageJson.version,
+            ...Object.fromEntries(
+                Object.entries(packageJson.dependencies || {}).filter(
+                    ([packageKey]) => packageRegex.test(packageKey)
+                )
+            ),
+            ...Object.fromEntries(
+                Object.entries(packageJson.devDependencies || {}).filter(
+                    ([packageKey]) => packageRegex.test(packageKey)
+                )
+            )
+        };
+        versionBanner = Object.entries(pwaStudioVersions)
+            .map(([packageKey, version]) => `${packageKey}: ${version}`)
+            .join(', ');
```

Comment:
```suggestion
        versionBanner = projectConfig.section('staging').buildId;
        if (!versionBanner || versionBanner.trim().length === 0) {
            const packageJson = require(path.resolve(context, './package.json'));
            const packageRegex = /^@magento|^@apollo/;
            const pwaStudioVersions = {
                'pwa-studio': packageJson.version,
                ...Object.fromEntries(
                    Object.entries(packageJson.dependencies || {}).filter(
                        ([packageKey]) => packageRegex.test(packageKey)
                    )
                ),
                ...Object.fromEntries(
                    Object.entries(packageJson.devDependencies || {}).filter(
                        ([packageKey]) => packageRegex.test(packageKey)
                    )
                )
            };
            versionBanner = Object.entries(pwaStudioVersions)
                .map(([packageKey, version]) => `${packageKey}: ${version}`)
                .join(', ');
        }    
```

Reply:
CCing in @supernova-at

Reply:
I'm fine keeping the build ID support in there but I'd prefer it not to be an either / or.

Output all the versions as we have here and if there's a build ID present, we can output that too 👍

I'm thinking about the case where someone accidentally leaves that environment variable set on their production deployment.

Reply:
I've added the build ID when available!